### ISSUE TITLE
Get Azure AD Application Owner (Get-AzADApplicationOwner)

### ISSUE THREAD

**Issue by JamesDLD** [MicrosoftDocs/feedback#1339]
**Is your feature request related to a problem? Please describe.**
Couldn't get Azure AD application owner(s) through the PowerShell module Az.Accounts (version 1.3.0).

**Describe the solution you'd like**
Would like to have a cmdlet called "Get-AzADApplicationOwner".

**Describe alternatives you've considered**
A property "Owner" of the Azure Ad application when we use the cmdlet "Get-AzADApplication".

**Additional context**
As we can use Get-AzADApplication (Az.Accounts module) instead of Get-AzureADApplication (Azure Ad module).
Why there is not a Get-AzADApplicationOwner (Az.Accounts module) cmdlet instead of Get-AzureADApplicationOwner (Azure Ad module).

**Comment by sudeepku** [comment]
@JamesDLD you can try  Get-AzureADApplicationOwner -ObjectId <ObjectId>
 see details here -https://docs.microsoft.com/en-us/powershell/module/azuread/get-azureadapplicationowner?view=azureadps-2.0

**Comment by JamesDLD** [comment]
@sudeepku
Additional context
As we can use Get-AzADApplication (Az.Accounts module) instead of Get-AzureADApplication (Azure Ad module).
Why there is not a Get-AzADApplicationOwner (Az.Accounts module) cmdlet instead of Get-AzureADApplicationOwner (Azure Ad module)

**Comment by g0pinath** [comment]
Same problem here, its making it difficult for reporting.
We cant go back to AzureRM module now, all other workbooks are now in Az.

**Comment by g0pinath** [comment]
> @sudeepku
> Additional context
> As we can use Get-AzADApplication (Az.Accounts module) instead of Get-AzureADApplication (Azure Ad module).
> Why there is not a Get-AzADApplicationOwner (Az.Accounts module) cmdlet instead of Get-AzureADApplicationOwner (Azure Ad module)

James - I ended up installing AzureAD module and I could use this cmdlet with Az module.

I think this command wasnt a part of AzureRm module to start with so may not come at all in Az.

https://docs.microsoft.com/en-us/powershell/module/azuread/get-azureadapplicationowner?view=azureadps-2.0 module says AzureAD

I agree with your point thoug, why not et-AzADApplicationOwner (Az.Accounts module) cmdlet instead of Get-AzureADApplicationOwner (Azure Ad module)

**Comment by JamesDLD** [comment]
Hello @g0pinath 
indeed both commands weren't part of the AzureRm module but as Get-AzADApplication became part of the Az module I am asking to go to the end of reasoning and publish the command **Get-AzADApplicationOwner**.

**Comment by dixit2d** [comment]
Is there any timeline for getting this API. Any other workround to get the application owner for a registered application?

**Comment by deepnetwork** [comment]
@RobdeJong, would love this request be considered for addition to the `AZ.Accounts` module.

**Comment by david-lpg** [comment]
Same for us, the functions Get-AzADApplicationOwner/Set-AzADApplicationOwner would really simplify work for us. The workaround is to use the old Get/Set-AzureADApplicationOwner (need to perform Connect-AzureAD first and the module AzureAD does not seem to work in powershell 7)

**Comment by andrewiankidd** [comment]
I've just hit this too, want to be able to add the service principal im using as an owner to an AD app, however I can't use 'Add-AzureADApplicationOwner' and there is no modern replacement it seems.

+1

**Comment by dorgrin6** [comment]
Until this get picked up, you can also use [AzureCli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest):
az ad app owner list --id <App Object Id>

**Comment by candidson** [comment]
Any update regarding this? Is there a timeline which could be communicated?

**Comment by kaiaschulz** [comment]
Hi,
since I had the same issue in past we decided to use the Azure Graph API.
In the end, the AZ. cmdlets are also using the Azure APIs.
Advantage is that we don't care about AZ versions anymore and are more independent.

### So, how can you easily get your AzureAD owned objects?
**Prerequisite is a bearer token which can be easily created and use with this code-snippet:**
[https://github.com/JulianHayward/AzAPICall/blob/main/AzAPICall.ps1](https://github.com/JulianHayward/AzAPICall/blob/main/AzAPICall.ps1)
_Beginning Line 610 I successfully replaced and tested the code:_
```
#create bearer token
createBearerToken -targetEndPoint "MSGraphAPI"

# https://docs.microsoft.com/en-us/graph/api/user-list-ownedobjects?view=graph-rest-1.0&tabs=http
$currentTask = "Getting all my AzureAD owned applications"
Write-Host "$currentTask"
$uri = "$(($htAzureEnvironmentRelatedUrls).($checkContext.Environment.Name).MSGraphUrl)/beta/me/ownedObjects"
$method = "GET"
$requestAllAzureAdOwnedObjects = ((AzAPICall -uri $uri -method $method -currentTask $currentTask))
$aadOwnedApplications = $requestAllAzureAdOwnedObjects | Where-Object { $_."@odata.type" -match "application" }

# https://docs.microsoft.com/en-us/graph/api/application-list-owners?view=graph-rest-1.0&tabs=http
$currentTask = "Getting all AzureAD application owner of specific application"
Write-Host "$currentTask"
$uri = "$(($htAzureEnvironmentRelatedUrls).($checkContext.Environment.Name).MSGraphUrl)/beta/applications/<AzureAD Application Object ID>/owners" #AzureAD Application Object ID
$method = "GET"
$requestAllAzureAdOwnerOfSpecificApp = ((AzAPICall -uri $uri -method $method -currentTask $currentTask))
$requestAllAzureAdOwnerOfSpecificApp.displayName
```

**Comment by justengraves** [comment]
Bump, would be convenient to have Get-AzADApplicationOwner as part of Az.Account or Az.Resources to more easily automate app registration auditing.

**Comment by PinewoodNut** [comment]
Running into the same issue.  Please add Get-AzADApplicationOwner / Set-AzADApplicationOwner to Az.Account.

**Comment by quix303** [comment]
Are there any updates or ETA on when this might be possible with Get-AzADApplicationOwner?  Missing features like this force flipping between modules and complicate scripting and other processes, or worse having to authenticate twice in both modules.

**Comment by LockTar** [comment]
I don't think it will come to the Az Module. Microsoft released a new PowerShell Module called Microsoft.Graph which will be the way forward. 
In this module the command is available. See the documentation for the  [Get-MgApplicationOwner](https://docs.microsoft.com/en-us/powershell/module/microsoft.graph.applications/get-mgapplicationowner?view=graph-powershell-1.0)  command.

**Comment by LockTar** [comment]
See for more information these articles and my comments beneath them:

- [Article 1](https://techcommunity.microsoft.com/t5/azure-tools/azure-ad-to-microsoft-graph-migration-for-azure-command-line/ba-p/2836666)
- [Article 2](https://techcommunity.microsoft.com/t5/azure-active-directory-identity/automate-and-manage-azure-ad-tasks-at-scale-with-the-microsoft/ba-p/1942489)
